StakingRewards/contracts/StakingRewards.sol

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+// Uncomment this line to use console.log
+// import "hardhat/console.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {IStakingRewards} from "./interfaces/IStakingRewards.sol";
+
+error ZeroAmount();
+error WaitToFinish();
+error NotEnoughBalance();
+error TooHighReward();
+error FailedToWithdrawStaking();
+
+/**
+ * Stakes tokens for a certain duration and gets rewards according to their
+ * staked shares
+ */
+contract StakingRewards is IStakingRewards, Ownable, Pausable, ReentrancyGuard {
+  using SafeERC20 for IERC20;
+
+  // Address to staking ERC20 token
+  IERC20 public stakingToken;
+  // Address to rewards ERC20 token
+  IERC20 public rewardsToken;
+  // Block timestamp of the end of reward session
+  uint256 public periodFinish;
+  // Reward rate
+  uint256 public rewardRate;
+  // Rewards duration in seconds
+  uint256 public rewardsDuration = 7 days;
```

Comment:
This contract update this variable in constructor function.
So don't need this initial value.